pkg/chunked/filesystem_linux.go

```diff
@@ -49,8 +49,31 @@ type fileMetadata struct {
 	skipSetAttrs bool
 }
 
+// splitPath takes a file path as input and returns two components: dir and base.
+// Differently than filepath.Split, this function handles some
+// edge cases.  If the path refers to a file in the root directory, the returned
+// dir is the empty string.
+// The returned base value is never empty, it never contains any slash and the
+// value "..".
+func splitPath(path string) (string, string) {
+	path = filepath.Clean(path)
+	dir, base := filepath.Split(path)
+	if isRootDirectory(dir) {
+		dir = ""
+		if base == ".." {
+			base = "."
+		}
+	} else {
+		dir = removeDotDotPrefix(filepath.Clean(dir))
```

Comment:
(I think the `filepath.Clean` here does nothing, but it doesn’t hurt.)

Reply:
was used as a lazy replacement for ` strings.TrimSuffix(path, "/")` and have a consistent result for the tests.  I'll replace it.